Code for the first assignment

JakobRoseke opened the pull request:
This code is for the first assignment, it is very simple but we might use it later on the project

JakobRoseke committed:
Code for the first assignment

This code is for the first assignment, it is very simple but we might use it later on the project

platisd commented on stopBeforeObstacle/stopBeforeObstacle.ino:
```diff
@@ -0,0 +1,49 @@
+    //Else we stop the car.
+    
+    if(front_sensor.getDistance() > 25 || front_sensor.getDistance() == 0){
+    
```
This will make two measurements which will not be necessarily the same.
What if the first call to getDistance returns 0 (e.g there is no obstacle nearby) and the second one 100?

JakobRoseke replied:
I'm not 100% sure I understand, do you mean that in the If loop we will get two instances/measurements of the distance? We tested the code and it worked the way we expected.

I guess we could create a variable, int distance = front_sensor.getDistance();, and then place that variable in the two conditions.

platisd replied:
> I guess we could create a variable, int distance = front_sensor.getDistance();, and then place that variable in the two conditions.

Exactly. 

> We tested the code and it worked the way we expected.

The fact that it happened to work, doesn't mean it always would. :)
This is why you should be writing many and short automated tests, instead of manually trying things out. :+1:

JakobRoseke replied:
Alright! So, what happens now? Should we remove the pull request and do change the code and then merge again before the end of the day? Or can we agree to the changes and change them if/when we need it?

JakobRoseke replied:
Ahhh, it looks like I can change the file here in GitHub, is that the next move?

platisd replied:
> Ahhh, it looks like I can change the file here in GitHub, is that the next move?

You can do that since it's a small change but generally you would just commit again on the same branch and push. Then this pull request would be updated.

JakobRoseke committed:
Update distance measurement

Changes to have the measurement for the distance was used, instead of getting two measurements there now is one that is used twice.
